Nodejs/Product/Nodejs/Intellisense/IntellisenseController.cs

```diff
@@ -1,663 +1,772 @@
+                                    return VSConstants.S_OK;
+                                }
+                            } else {
+                                _activeSession.Dismiss();
+                            }
+                            break;
+                        case VSConstants.VSStd2KCmdID.TAB:
+                            if (!_activeSession.IsDismissed) {
+                                _activeSession.Commit();
+                                return VSConstants.S_OK;
+                            }
+                            break;
+                        case VSConstants.VSStd2KCmdID.BACKSPACE:
+                        case VSConstants.VSStd2KCmdID.DELETE:
+                        case VSConstants.VSStd2KCmdID.DELETEWORDLEFT:
+                        case VSConstants.VSStd2KCmdID.DELETEWORDRIGHT:
+                            int res = _oldTarget.Exec(ref pguidCmdGroup, nCmdID, nCmdexecopt, pvaIn, pvaOut);
+                            if (_activeSession != null && !_activeSession.IsDismissed) {
+                                _activeSession.Filter();
+                            }
+                            return res;
+                    }
+                }
+            } else if (_sigHelpSession != null) {
+                if (pguidCmdGroup == VSConstants.VSStd2K) {
+                    switch ((VSConstants.VSStd2KCmdID)nCmdID) {
+                        case VSConstants.VSStd2KCmdID.BACKSPACE:
+                            bool fDeleted = Backspace();
+                            if (fDeleted) {
+                                return VSConstants.S_OK;
+                            }
+                            break;
+                        case VSConstants.VSStd2KCmdID.LEFT:
+                            _editOps.MoveToPreviousCharacter(false);
+                            UpdateCurrentParameter();
+                            return VSConstants.S_OK;
+                        case VSConstants.VSStd2KCmdID.RIGHT:
+                            _editOps.MoveToNextCharacter(false);
+                            UpdateCurrentParameter();
+                            return VSConstants.S_OK;
+                        case VSConstants.VSStd2KCmdID.HOME:
+                        case VSConstants.VSStd2KCmdID.BOL:
+                        case VSConstants.VSStd2KCmdID.BOL_EXT:
+                        case VSConstants.VSStd2KCmdID.EOL:
+                        case VSConstants.VSStd2KCmdID.EOL_EXT:
+                        case VSConstants.VSStd2KCmdID.END:
+                        case VSConstants.VSStd2KCmdID.WORDPREV:
+                        case VSConstants.VSStd2KCmdID.WORDPREV_EXT:
+                        case VSConstants.VSStd2KCmdID.DELETEWORDLEFT:
+                            _sigHelpSession.Dismiss();
+                            _sigHelpSession = null;
+                            break;
+                    }
+                }
+            } else {

```

Comment:
Is there a reason you added this else statement here? It changes the ctrl flow unnecessarily and has potential to introduce subtle bugs.